java-manta-client-unshaded/src/main/java/com/joyent/manta/client/crypto/MantaEncryptedObjectInputStream.java

```diff
@@ -654,13 +654,16 @@ private void readRemainingBytes() throws IOException {
     /**
      * Calculates the size of buffer to use when reading chunks of bytes based on the known
      * content length.
+     *
      * @return size of buffer to read into memory
+     * @param contentLength the content
+     * @param cipherDetails the cipher in use
      */
-    private int calculateBufferSize() {
-        long cipherTextContentLength = ObjectUtils.firstNonNull(getContentLength(), -1L);
+    static int calculateBufferSize(final Long contentLength, final SupportedCipherDetails cipherDetails) {
+        long cipherTextContentLength = ObjectUtils.firstNonNull(contentLength, -1L);
 
         if (cipherTextContentLength >= 0) {
-            cipherTextContentLength -= this.cipherDetails.getAuthenticationTagOrHmacLengthInBytes();
+            cipherTextContentLength -= cipherDetails.getAuthenticationTagOrHmacLengthInBytes();
```

Comment:
It is a little odd that you removed `this` here, but added `this` in your other changes.

Reply:
This method was made static and package-private so that it can be used from `MantaEncryptedObjectInputStreamAutoFailureRecoveryTest` to inject failures relative to the buffer size: https://github.com/joyent/java-manta/pull/426/files#diff-58132a0f33da576d1b7e967879396e57R230

Reply:
:+1: I didn't catch that detail from the diff. Thanks!

Reply:
Oh, if it is static, then we should use the class name to access it. That's what was tripping up my eyes.

Reply:
Added class name to clarify static access.